tests/test_form_fields.py

```diff
@@ -0,0 +1,326 @@
+        for checkbox in checkboxes:
+            assert checkbox.get("disabled") == ("disabled" if disabled else None)
+
+
+class TestRDURadioField:
+    class FormForTest(FlaskForm):
+        radio_field = RDURadioField(label="radio_field", choices=[(1, "one"), (2, "two"), (3, "three")])
+        radio_field_invalid = RDURadioField(
+            label="radio_field",
+            choices=[(1, 1), (2, 2), (3, 3)],
+            validators=[DataRequired(message="failed validation")],
+        )
+        other_field = RDUStringField(label="other_field")
+
+    def setup(self):
+        self.form = self.FormForTest()
+
+    def teardown(self):
+        self.form = None
+
+    def test_legend_is_rendered(self):
+        doc = html.fromstring(self.form.radio_field())
+
+        assert doc.xpath("//legend")
+        assert "radio_field" in doc.xpath("//legend")[0].text
+
+    def test_radio_choices_are_rendered(self):
+        doc = html.fromstring(self.form.radio_field())
+
+        assert len(doc.xpath("//input[@type='radio']")) == 3
+
+    def test_radio_choices_have_correct_values(self):
+        doc = html.fromstring(self.form.radio_field())
+
+        radios = doc.xpath("//input[@type='radio']")
+        for i, radio in enumerate(radios):
+            assert radio.get("value") == str(self.form.radio_field.choices[i][0])
+
+    def test_radio_labels_are_rendered(self):
+        doc = html.fromstring(self.form.radio_field())
+
+        radios = doc.xpath("//input[@type='radio']")
+        for i, radio in enumerate(radios):
+            label = radio.xpath("following-sibling::label")[0]
+            assert label.text == self.form.radio_field.choices[i][1]
+
+    @pytest.mark.parametrize("disabled", [True, False])
+    def test_render_field_with_disabled_causes_all_inputs_to_be_disabled(self, disabled):
+        doc = html.fromstring(self.form.radio_field(disabled=disabled))
+
+        radios = doc.xpath("//input[@type='radio']")
+        for radio in radios:
+            assert radio.get("disabled") == ("disabled" if disabled else None)
+
+    def test_other_field_is_rendered(self):
+        self.form.radio_field.set_other_field(self.form.other_field)
+        doc = html.fromstring(self.form.radio_field())
+
+        assert len(doc.xpath("//input[@type='text']")) == 1
+
+    def test_other_field_includes_show_hide_script(self):
+        self.form.radio_field.set_other_field(self.form.other_field)
+        doc = html.fromstring(self.form.radio_field())
+
+        assert "showHideControl" in doc.xpath("//script")[0].text
+
+
+class TestRDUStringField:
+    class FormForTest(FlaskForm):
+        string_field = RDUStringField(label="string_field", hint="string_field hint")
+        string_field_invalid = RDUStringField(
+            label="string_field", hint="string_field hint", validators=[DataRequired(message="failed validation")]
+        )
+
+    def setup(self):
+        self.form = self.FormForTest()
+
+    def teardown(self):
```

Comment:
Can probably kill this teardown?  Or remove the `#` if we really want to reset form each time.

Reply:
Done. 👍